### PR TITLE
[WIP][SYCL] update event profiling support with the new APIs.

### PR DESCRIPTION
**Pull request by rbegam**
Signed-off-by: rbegam <rehana.begam@intel.com>

### CODE REVIEW

**Comment by smaslov-intel** [comment]
tag @jandres742

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -2359,29 +2359,30 @@ pi_result piEventGetProfilingInfo(pi_event Event, pi_profiling_info ParamName,
...
+
   switch (ParamName) {
   case PI_PROFILING_INFO_COMMAND_START: {
-    uint64_t ContextStart;
-    ZE_CALL(zeEventGetTimestamp(
-        Event->ZeEvent, ZE_EVENT_TIMESTAMP_CONTEXT_START, &ContextStart));
-    ContextStart *= ZeTimerResolution;
-    return ReturnValue(uint64_t{ContextStart});
+    ze_kernel_timestamp_result_t* tsResult = nullptr;
+    ZE_CALL(zeCommandListAppendQueryKernelTimestamps(
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:2368]: The spec says https://spec.oneapi.com/level-zero/latest/core/api.html?highlight=zecommandlistappendquerykerneltimestamps#_CPPv440zeCommandListAppendQueryKernelTimestamps24ze_command_list_handle_t8uint32_tP17ze_event_handle_tPvPK6size_t17ze_event_handle_t8uint32_tP17ze_event_handle_t:

> Appends a query of an events’ timestamp value(s) into a command list.

I don't see how this is going to achieve what's needed. It looks like these timestamp query should be appended around the command being profiled. Could you explain how this is working?

**Comment by bader** [comment]
@rbegam, ping.

**Comment by rbegam** [comment]
taking a look. it needs some following up.

**Comment by rbegam** [comment]
the support has been included, closing this as a duplicate.

**Comment by bader** [comment]
Duplicate of https://github.com/intel/llvm/commit/6d34b95c3eb3bccf5d739d66a7c57eed196cc206.

**Pull request closed without merging**